File: 3255. Find the Power of K-Size Subarrays II.java
class Solution {
    public int[] resultsArray(int[] nums, int k) {
        if (k == 1) return nums;
        int[] result = new int[nums.length-k+1];
        int left = 0;
        for (int right = 1; right < nums.length; right++) {
            if (nums[right] != nums[right-1]+1) {
                while (left < nums.length-k+1 && left != right) {
                    result[left++] = -1;
                }
            }
            if (right - left + 1 == k) {
                result[left++] = nums[right];
            }
        }
        return result;
    }
}

File: go/3255. Find the Power of K-Size Subarrays II.go
func resultsArray(nums []int, k int) []int {
    if k == 1 { return nums }
    result := make([]int, len(nums)-k+1)
    left := 0
    for right := 1; right < len(nums); right++ {
        if nums[right] != nums[right-1]+1 {
            for left < len(nums)-k+1 && left != right {
                result[left] = -1
                left++
            }
        }
        if right - left + 1 == k {
            result[left] = nums[right]
            left++
        }
    }
    return result
}

